sdk/daml-script/test/daml/UpgradeTestLib.daml

```diff
@@ -19,28 +20,48 @@ import DA.Assert
 import DA.Foldable
 import DA.Time
 
+import DA.Text qualified as T
+
 participant0 : ParticipantName
 participant0 = ParticipantName "participant0"
 
 participant1 : ParticipantName
 participant1 = ParticipantName "participant1"
 
-type TestTree = Text -> Script ()
+data RunMode
+  = IdeLedger
+  | Canton
+  deriving (Show)
+
+data TestState = TestState
```

Comment:
Ah I like this more, this would also let us pass more data down to the script if we wanted to in future - nice!

Reply:
I do wonder if we should be using `with` though ;)